src/azul/bigquery_reservation.py

```diff
@@ -0,0 +1,141 @@
+from typing import (
+    Optional,
+)
+
+import attr
+from google.cloud.bigquery_reservation_v1 import (
+    Assignment,
+    CapacityCommitment,
+    Reservation,
+    ReservationServiceClient,
+)
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    RequirementError,
+    cached_property,
+    logging,
+    require,
+)
+from azul.terra import (
+    TerraClient,
+)
+
+log = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True, kw_only=True, frozen=True)
+class ResourceNames:
```

Comment:
There's a smell here in that there ever will be one set of resource names. I think you should track those three resource names as normal attributes and avoid passing this de-facto singleton around. That seems like the more straight forward implementation. I could be convinced otherwise, either by example or reasoning.